### PR TITLE
feat(build): add cmake build system

### LINKED ISSUE #17: CMake support
a modern sdk should have cmake support, because vcxproj is xml bullshit.

### PR DESCRIPTION
**Pull request by kehrazy**
this adds cmake support via cmkr

### CODE REVIEW

**Comment by es3n1n** [comment]
Looks promising, thanks. I'll check it as soon as i can.
Also maybe it would be cool to move vcxproj and other stuff to a separate folder.

**Comment by kehrazy** [comment]
> Also maybe it would be cool to move vcxproj and other stuff to a separate folder.

sure.

and about minhook/imgui dependencies - we can use vcpkg/fetchcontent here - cmkr handles it beautifully, although that would fuck up the visual studio projects.

File: readme.md [hunk 1]
```diff
@@ -26,5 +26,13 @@ Any contributions you make are **greatly appreciated**.
 4. Push to the Branch (`git push origin feature/AmazingFeature`)
 5. Open a Pull Request
 
+## Building
```
Comment (kehrazy) [inline readme.md:28]: this might need to go to the `CONTRIBUTING.md` file later though

File: cmake/cmkr.cmake [hunk 2]
```diff
@@ -0,0 +1,253 @@
+include_guard()
```
Comment (kehrazy) [inline cmake/cmkr.cmake:1]: sucks, but cmkr explicitly says that the CMakeFiles.txt is intended to be included in the repo

File: readme.md (hunk 1 above)
Comment (es3n1n) [reply readme.md:28]: yeah but not really, for now all building guides are small enough to fit in the readme imo

**Comment by es3n1n** [comment]
Initially i thought about moving vcxproj files to msvc dir too, but when i did so it broke the `ShowAllFiles` option. After 15 minutes in google i didn't find the guide on how to change the directory that it uses by default when showing all files. I think we'll leave vcxproj files in the project directories.

> and about minhook/imgui dependencies - we can use vcpkg/fetchcontent here - cmkr handles it beautifully, although that would fuck up the visual studio projects.

i think we'll leave everything as it is for now, ideally i wanna add all dependencies as submodules later, cba for now


Everything else looks good to me, gonna merge it now.

**Pull request merged**